transit/rest_api/urls.py

```diff
@@ -8,17 +8,35 @@
 from rest_framework_simplejwt.views import TokenRefreshView, TokenVerifyView, TokenObtainPairView
 
 
+from transit.rest_api.permissions import IsUserManager
+
+
 class UserSerializer(serializers.HyperlinkedModelSerializer):
+    class Meta:
+        model = User
+        fields = ['url', 'username', 'email', 'is_staff', 'password']
+        extra_kwargs = {'password': {'write_only': True}}
+
+    def create(self, validated_data):
+        user = User(**validated_data)
+        user.set_password(validated_data['password'])
+        user.save()
+        return user
+
+
+class UserUpdateSerializer(serializers.HyperlinkedModelSerializer):
     class Meta:
         model = User
         fields = ['url', 'username', 'email', 'is_staff']
 
 
 # ViewSets define the view behavior.
 class UserViewSet(viewsets.ModelViewSet):
-    permission_classes = (IsAuthenticated,)
+    permission_classes = (IsAuthenticated, IsUserManager)
     queryset = User.objects.all()
-    serializer_class = UserSerializer
+
+    def get_serializer_class(self):
+        return UserUpdateSerializer if self.request.method == 'PUT' else UserSerializer
```

Comment:
Is this approach compatible with swagger?

Reply:
yes. It is compatible
![Screenshot from 2022-08-11 10-38-05](https://user-images.githubusercontent.com/52816247/184095623-de021919-4d89-462e-8a60-e33f9ddbc7c1.png)
![Screenshot from 2022-08-11 10-38-20](https://user-images.githubusercontent.com/52816247/184095627-e7e006be-d7df-448c-ada7-35e8f3cc8863.png)

Reply:
That's fantastic, thanks